src/cloupe_package_nj9/cloupe.py

```diff
@@ -0,0 +1,268 @@
+            f.seek(start)
+            read_size = end - start
+            block = f.read(read_size)
+            if block[:2] == b"\x1f\x8b":
+                logging.debug("compressed block")
+                block = self.decompress(block)
+            if as_json:
+                return json.loads(block.decode("utf-8", errors="replace"))
+            return block
+
+
+if __name__=="__main__":
+    logging.basicConfig(level=logging.INFO, format="[%(asctime)s][%(levelname)s] %(message)s")
+    # Create the parser
+    parser = argparse.ArgumentParser(description='Cloupe parser which extracts Barcodes, Features and Annotations info')
+    # Add arguments
+    parser.add_argument('input_file', type=str, help='Path to the input file')
+    # Parse the arguments
+    args = parser.parse_args()
+    # Access the arguments
+    # Load cloupe file
+    cwd = os.getcwd()
+    # hardcoding the location of the files that are generated out of the script
+    barcodes = cwd + '/barcodes.csv'
+    features = cwd + '/features.csv'
+    barcode_clusters = cwd + '/annotations.csv'
+    cloupe = Cloupe(args.input_file)
+
+    count_values = {}
+    # Getting the count of Features and Barcodes to set the limits for the columns in the csv files
+    for prop in ['FeatureCount', 'BarcodeCount']:
+        count_values[prop] = cloupe.matrices[0][prop]
+
+
+    collection = []
+    # writing the barcodes to the csv files
+    with open(barcodes, 'w') as barcodes_file:
+        barcodes_file.write("Barcodes\n")
+        barcodes_file.write(
+            "\n".join(
+            cloupe.matrices[0]['Barcodes'][:count_values['BarcodeCount']]
+            )
+        )# Efficient one line code
+
+    with open(features, 'w') as features_file:
+        features_file.write("FeatureIds,FeatureNames\n")
+        features_file.write(
+            "\n".join(
+                [
+                    re.sub(r"[()' ]", "", str(pair))  # Remove parentheses, single quotes, and spaces
+                    for pair in zip(
+                    cloupe.matrices[0]["FeatureIds"][: count_values["FeatureCount"]],
+                    cloupe.matrices[0]["FeatureNames"][: count_values["FeatureCount"]],
+                )
+                ]
+            )
+        )# Using zip command along with list comprehension to convert the two list into strings for the csv as elements
+
+    # Plotting the info
+    spatial_embedding = cloupe.projections['Spatial']
+    plt.scatter(x=spatial_embedding[0], y=spatial_embedding[1])
+    plt.gca().invert_yaxis()
+    plt.show()
+
+    # Final csv addition
+    with open(barcode_clusters, 'w') as barcode_clusters_file:
+        barcode_clusters_file.write("Barcodes,{},{}\n".format(cloupe.celltracks[0]["Name"],cloupe.celltracks[1]["Name"]))
+        barcode_clusters_file.write(
+            "\n".join(
+                [
+                    re.sub(r"[()' ]", "", str(pair))  # Remove parentheses, single quotes, and spaces
```

Comment:
This is also fixed to only support 2 celltracks. 
Should be dynamic based on the number of items in `cloupe.celltracks`.

`re.sub(r"[()' ]", "", str(pair))` is very odd